### PR TITLE
msys2,docs, clang-tidy

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will fix some clang-tidy issues in some core files, fix the msys2 build, and documentation builds

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -408,7 +408,8 @@ if(MSYS OR CYGWIN)
     if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
         set(ENABLE_IPC_CORE FALSE)
     endif()
-    target_compile_definitions(helics_base INTERFACE -D_WIN32_WINNT=0x0601)
+    # setting to windows 10 for all compilation
+    add_compile_definitions(_WIN32_WINNT=0x0A00)
```
Comment (nightlark) [inline CMakeLists.txt:412]: Does this mean that HELICS will only work on Windows 10 or newer when building under MSYS/Cygwin (e.g. dropping support for older versions)?

File: src/helics/application_api/helicsPrimaryTypes.cpp [hunk 2]
```diff
@@ -34,6 +34,13 @@ static const std::set<std::string> falseString{"0",
                                                "false",
                                                "False",
                                                "FALSE",
+                                               "OFF",
+                                               "Disabled",
+                                               "disable",
+                                               "disabled",
+                                               "DISABLE",
+                                               "off"
+                                               "DISABLED",
```
Comment (nightlark) [inline src/helics/application_api/helicsPrimaryTypes.cpp:43]:
```suggestion
                                               "off",
                                               "Off",
                                               "OFF",
                                               "disabled",
                                               "Disabled",
                                               "DISABLED",
                                               "disable",
                                               "Disable",
                                               "DISABLE",
```
Include all capitalization variants, and sort so it is easier to tell what is there/missing.

File: config/cmake/compiler_flags.cmake [hunk 3]
```diff
@@ -119,13 +119,18 @@ if(${PROJECT_NAME}_ENABLE_EXTRA_COMPILER_WARNINGS)
...
+        endif()
     endif()
     if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
         target_compile_options(compile_flags_target INTERFACE $<$<COMPILE_LANGUAGE:CXX>:-Wshadow>)
         target_compile_options(
             compile_flags_target INTERFACE -Wdocumentation -Wno-documentation-deprecated-sync
         )
-        if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 13.0)
+        if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0)
             message(STATUS "clang>13")
```
Comment (nightlark) [inline config/cmake/compiler_flags.cmake:134]:
```suggestion
            message(STATUS "clang>=13")
```

File: config/cmake/CheckLatestCXXStandardOption.cmake [hunk 4]
```diff
@@ -21,6 +21,12 @@ if(CMAKE_CXX_STANDARD LESS 17)
     message(FATAL_ERROR "{$PROJECT_NAME} requires C++17 or greater")
 endif()
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13.0)
+    if(CMAKE_CXX_STANDARD LESS 20)
+        message(FATAL_ERROR " if using gcc>=13.0 {$PROJECT_NAME} requires C++20 or greater")
```
Comment (nightlark) [inline config/cmake/CheckLatestCXXStandardOption.cmake:26]:
```suggestion
        message(FATAL_ERROR " using gcc>=13.0 {$PROJECT_NAME} requires setting standard to C++20 or greater")
```

File: CMakeLists.txt (hunk 1 above)
Comment (phlptp) [reply CMakeLists.txt:412]: It means the default is windows 10,  Maybe we want to make a way to override this definition for some odd situations?

**Comment by phlptp** [comment]
@nightlark  any other changes I should make in this one?

**Review by nightlark — APPROVED** [review]

**Pull request merged**